### PR TITLE
move theia app to gitpod-io/theia-app

### PR DESCRIPTION
**Pull request by AlexTugarev**
Theia based IDE is moving to https://github.com/gitpod-io/theia-app

Test with https://at-ide-theia.staging.gitpod-dev.com/workspaces/

Tasks
 - [x] copy builtin plugins
   Screen Shot 2021-03-04 at 13 49 53
 - [x] apply theia patches

closes https://github.com/gitpod-io/gitpod/issues/3296

### CODE REVIEW

File: components/ide/theia/leeway.Dockerfile [hunk 1]
```diff
@@ -24,47 +23,61 @@ RUN apk add --no-cache bash gcc g++ make pkgconfig python libc6-compat libexecin
...
 COPY --from=builder_alpine /theia/ /theia/
 
+
 # standard supervisor entrypoint used when supervisor isn't coming from this image
 WORKDIR /ide/
 COPY supervisor-ide-config.json /ide/
 
-ENV GITPOD_BUILT_IN_PLUGINS /theia/node_modules/@gitpod/gitpod-ide/plugins/
-COPY components-theia-app--builtin-plugins/plugins/ ${GITPOD_BUILT_IN_PLUGINS}
+# ENV GITPOD_BUILT_IN_PLUGINS /theia/node_modules/@gitpod/gitpod-ide/plugins/
```
Comment (AlexTugarev) [inline components/ide/theia/leeway.Dockerfile:77]: TBD

File: components/ide/theia/leeway.Dockerfile [hunk 2]
```diff
@@ -24,47 +23,61 @@ RUN apk add --no-cache bash gcc g++ make pkgconfig python libc6-compat libexecin
...
-RUN yarn patch-package
-
 # copy native dependencies of node modules
-RUN find /theia/node_modules/ -iname *.node -exec package-libs.sh {} \;
+RUN find /theia/theia-app/node_modules/ -iname *.node -exec package-libs.sh {} \;
 
 RUN cp /theia/node/bin/node /theia/node/bin/gitpod-node && rm /theia/node/bin/node
 
+# patching plugin ext host process to fix css extension compatibility, see https://github.com/gitpod-io/gitpod/pull/2483
+# RUN cp -r /theia/theia-app/node_modules/@gitpod/gitpod-ide/patches /theia/theia-app/patches
```
Comment (AlexTugarev) [inline components/ide/theia/leeway.Dockerfile:61]: TBD

Comment (akosyakov) [reply components/ide/theia/leeway.Dockerfile:61]: It should be moved and kept.

Comment (AlexTugarev) [reply components/ide/theia/leeway.Dockerfile:61]: fully agree, see the todo from ☝🏻 
just, ditching leeway build led to different directory structures, I then took a shortcut and left a todo.

File: components/ide/theia/leeway.Dockerfile [hunk 3]
```diff
@@ -18,53 +17,73 @@ RUN mkdir -p /theia/node/bin \
...
 
+
 # cli config
 COPY bin /ide/bin
 RUN chmod -R ugo+x /ide/bin
 
 FROM scratch
 COPY --from=builder_alpine /theia/ /theia/
 
+ENV GITPOD_ENV_SET_GITPOD_BUILT_IN_PLUGINS /theia/theia-app/plugins/
```
Comment (AlexTugarev) [inline components/ide/theia/leeway.Dockerfile:80]: @akosyakov, that did work eventually :-)

Comment (akosyakov) [reply components/ide/theia/leeway.Dockerfile:80]: test

**Review by akosyakov — APPROVED** [review]
It works nicely and code changes looks good to me as well. Thank you!

**Comment by AlexTugarev** [comment]
@akosyakov thank YOU!
I'm merging now to keep moving.

**Comment by gtsiolis** [comment]
Thank you @AlexTugarev for this!

Screenshot 2021-03-05 at 4 33 44 PM

**Pull request merged**